File: Day26/DSA/Ransom_note.java
Question:- https://leetcode.com/problems/ransom-note/

Code:- 

class Solution {
    public boolean canConstruct(String ransomNote, String magazine) {
        HashMap<Character,Integer> map = new HashMap<>();

        for(char i : ransomNote.toCharArray())
        {
            int count = map.getOrDefault(i , 0);
            map.put(i , count + 1);
        }

        for(char j : magazine.toCharArray())
        {
            int count = map.getOrDefault(j , 0);
            map.put(j , count - 1);
        }

        for(Map.Entry<Character, Integer> ele : map.entrySet())
        {
            if(ele.getValue() > 0)
            {
                return false;
            }
        }

        return true;

    }
}

File: Day26/DSA/Ransom_note.cpp
Code:- 

class Solution {
public:
    bool canConstruct(string ransomNote, string magazine) {
        unordered_map<int, int> mp;
        for (auto i : ransomNote) {
            mp[i]++;
        }
        for (auto x : magazine) {
            mp[x]--;
        }

        for (auto i : mp) {
            if (i.second > 0)
                return false;
        }
        return true;
    }
};